sdk/src/Core/Amazon.Runtime/Pipeline/RetryHandler/RetryPolicy.cs

```diff
@@ -421,50 +425,59 @@ private bool IsClockskew(IExecutionContext executionContext, Exception exception
+            var correctedNow = CorrectClockSkew.GetCorrectedUtcNowForEndpoint(endpoint);
+            // Try getting server time from the headers
+            bool serverTimeDetermined = TryParseDateHeader(ase, out serverTime);
+
+            // If that fails, try to parse it from the exception message
+            if (!serverTimeDetermined)
+                serverTimeDetermined = TryParseExceptionMessage(ase, out serverTime);
+            serverTime = serverTime.ToUniversalTime();
+            // adjust the clock and always retry
+            var shouldRetry = AWSConfigs.CorrectForClockSkew && !AWSConfigs.ManualClockCorrection.HasValue;
+            if (isDefiniteClockSkewErrorCode && serverTimeDetermined)
             {
-                var endpoint = executionContext.RequestContext.Request.Endpoint.ToString();
-                var realNow = AWSConfigs.utcNowSource();
-
-                DateTime serverTime;
-
-                // Try getting server time from the headers
-                bool serverTimeDetermined = TryParseDateHeader(ase, out serverTime);
-
-                // If that fails, try to parse it from the exception message
-                if (!serverTimeDetermined)
-                    serverTimeDetermined = TryParseExceptionMessage(ase, out serverTime);
-
-                if (serverTimeDetermined)
+                Logger.InfoFormat(clockSkewMessageFormat,
+                    realNow, correctedNow, clientConfig.ClockOffset, serverTime, endpoint);
+                var newCorrection = serverTime - realNow;
+                CorrectClockSkew.SetClockCorrectionForEndpoint(endpoint, newCorrection);
+                if (shouldRetry)
                 {
-                    // using accurate server time, calculate correction if local time is off
-                    serverTime = serverTime.ToUniversalTime();
-                    var correctedNow = CorrectClockSkew.GetCorrectedUtcNowForEndpoint(endpoint);
-                    var diff =  correctedNow - serverTime;
-                    var newCorrection = serverTime - realNow;
-                    var absDiff = diff.Ticks < 0 ? -diff : diff;
+                    Logger.InfoFormat(clockSkewUpdatedFormat, newCorrection, endpoint);
+                    executionContext.RequestContext.IsSigned = false;
+                    return true;
+                }
+            }
+            // Some services such as S3 will return a generic 403 error for clock skew errors on HEAD requests
+            // with no error code. In this case we will always retry the request if the clock skew is greater than
+            // the threshold.
+            if ((isPossibleClockSkewErrorCode || isHead) && serverTimeDetermined)
+            {
+                DateTime clientTime = executionContext.RequestContext.Request.SignedAt ?? correctedNow;
+                var diff = clientTime - serverTime;
+                var absDiff = diff.Ticks < 0 ? -diff : diff;
```

Comment:
Is this a micro performance optimization instead of calling `Math.Abs`?

Reply:
I'm not sure if this gives a performance benefit, but I'd rather leave it this way since it was here before I made this change.

Reply:
Ok, I see it was in the old code as well.